src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs

```diff
@@ -11,35 +11,32 @@ namespace Microsoft.ML
     /// <summary>
     /// Specifies input and output column names for a transformation.
     /// </summary>
-    [BestFriend]
-    internal sealed class ColumnOptions
+    public sealed class InputOutputColumnPair
     {
-        private readonly string _outputColumnName;
-        private readonly string _inputColumnName;
+        /// <summary>
+        /// Name of the column to transform. If set to <see langword="null"/>, the value of the <see cref="OutputColumnName"/> will be used as source.
+        /// </summary>
+        public readonly string InputColumnName;
```

Comment:
>InputColumnName [](start = 31, length = 15)

Shouldn't it be in reverse - the input be set, but output be optional and equal to input?

Reply:
As per [https://github.com/dotnet/machinelearning/issues/2064](https://github.com/dotnet/machinelearning/issues/2064) we use the `outputColumnName` as `inputColumnName` when `inputColumnName` is null. We are doing this across the code base.

---
In reply to: [268727730](https://github.com/dotnet/machinelearning/pull/3066#discussion_r268727730) [](ancestors = 268727730)